--trace-fst causes internal error on unpacked array with shifted index assignment

When using `--trace-fst`, Verilator hits an internal error if you assign to an unpacked array with shifted indices. Without `--trace-fst`, everything works fine. 

## Code Snippet

```systemverilog
module Test #(
  localparam int DIM1 = 4,
  localparam int DIM2 = 8,
  localparam int SHIFT = 2
) (
  input logic [DIM1-1:0][DIM2-1:0] in
);

  logic [DIM1-1:0][DIM2+SHIFT-1:SHIFT] internal;

  for (genvar dim1 = 0; dim1 < DIM1; dim1++) begin
    for (genvar dim2 = 0; dim2 < DIM2; dim2++) begin
      assign internal[dim1][dim2+SHIFT] = in[dim1][dim2];
    end
  end

  for (genvar dim1 = 0; dim1 < DIM1; dim1++) begin
    for (genvar dim2 = 0; dim2 < DIM2; dim2++) begin
      always_ff @(negedge internal[dim1][dim2+SHIFT]) begin
        $display("%0b", internal[dim1][dim2+SHIFT]);
      end
    end
  end

endmodule
```

Use this command:

```bash
verilator test.sv --trace-fst --cc --exe --binary --main --top Test
```

It only crashes if `--trace-fst` is used. Without it, no error.

Here’s the error Verilator throws:

```
%Error: Internal Error: test.sv:9:40: ../V3FuncOpt.cpp:228: Inconsistent assignment
   9 |   logic [DIM1-1:0][DIM2+SHIFT-1:SHIFT] internal;
     |                                        ^~~~~~~~
```

## Version

Verilator 5.036 (2025-04-27)
Ubuntu 22.04
Always reproducible with `--trace-fst`

Add another issue 😛 

```
ver___024root__DepSet_h96c49fed__23.cpp: In function 'void ver___024root___eval_triggers__act__14(ver___024root*)':
ver___024root__DepSet_h96c49fed__23.cpp:4753:46: error: '__Vtrigcurrexpr_h1cad7197__0' was not declared in this scope
 4753 |     vlSelfRef.__Vtrigprevexpr_h1cad7197__0 = __Vtrigcurrexpr_h1cad7197__0;
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Looks like the SenExprBuilder is emitted to different cpp files. I'm almost certain this breaks after #5798. Maybe it's the same issue?

--- a/src/V3SplitVar.cpp
+++ b/src/V3SplitVar.cpp
@@ -919,14 +919,8 @@ public:
             points.emplace_back(ref.lsb(), false);  // Start of a region
             points.emplace_back(ref.msb() + 1, true);  // End of a region
         }
-        int bit_hi, bit_lo;
-        if (basicp() == dtypep) {
-            bit_hi = basicp()->hi();
-            bit_lo = basicp()->lo();
-        } else {  // packed struct, packed array. lo is 0
-            bit_hi = dtypep->width() - 1;
-            bit_lo = 0;
-        }
+        const int bit_lo = basicp()->lo();
+        const int bit_hi = bit_lo + dtypep->width() - 1;
         if (skipUnused && !m_rhs.empty()) {  // Range to be read must be kept, so add points here
             int lsb = bit_hi + 1;
             int msb = bit_lo - 1;

--- a/test_regress/t/t_split_var_types.v
+++ b/test_regress/t/t_split_var_types.v
@@ -10,13 +10,20 @@ module t(/*AUTOARG*/
     );
     input clk;
 
+    logic [7:0] data = 0;
     // Test loop
     always @ (posedge clk) begin
-        $write("*-* All Finished *-*\n");
-        $finish;
+       if (data != 15) begin
+          data <= data + 8'd1;
+       end else begin
+          $write("*-* All Finished *-*\n");
+          $finish;
+       end
     end
 
+
     bug5782 u_bug5782(.data_out());
+    bug5984 u_bug5984(.in(data));
 
 endmodule
 
@@ -28,4 +35,26 @@ module bug5782 (
     always begin
         data_out = data[7];
     end
+endmodule
+
+// #5984 inconsistent assignment due to wrong bit range calculation.
+module bug5984 (
+   input logic [1:0][3:0] in
+   );
+
+   logic [1:0][5:2] internal;
+
+   for (genvar dim1 = 0; dim1 < 2; dim1++) begin
+      for (genvar dim2 = 0; dim2 < 4; dim2++) begin
+         assign internal[dim1][dim2+2] = in[dim1][dim2];
+      end
+   end
+
+   for (genvar dim1 = 0; dim1 < 2; dim1++) begin
+      for (genvar dim2 = 0; dim2 < 4; dim2++) begin
+         always_ff @(negedge internal[dim1][dim2+2]) begin
+            $display("%0b", internal[dim1][dim2+2]);
+         end
+      end
+   end
 endmodule

